programs/ziti-edge-tunnel/ziti-edge-tunnel.c

```diff
@@ -406,16 +485,16 @@ static bool check_send_notification(tunnel_identity *tnl_id) {
 
 static notification_message *create_notification_message(tunnel_identity *tnl_id) {
     notification_message *notification = calloc(1, sizeof(struct notification_message_s));
-    notification->Message = malloc(MAXMESSAGELEN);
+    notification->Message = calloc(MAXMESSAGELEN, sizeof(char));
     if (tnl_id->MfaMaxTimeoutRem == 0) {
-        snprintf(notification->Message, MAXMESSAGELEN, "All of the services of identity %s have timed out", tnl_id->Name);
+        snprintf(notification->Message, MAXMESSAGELEN-1, "All of the services of identity %s have timed out", tnl_id->Name);
         notification->Severity = event_severity_critical;
     } else if (tnl_id->MfaMinTimeoutRem == 0) {
-        snprintf(notification->Message, MAXMESSAGELEN, "Some of the services of identity %s have timed out", tnl_id->Name);
+        snprintf(notification->Message, MAXMESSAGELEN-1, "Some of the services of identity %s have timed out", tnl_id->Name);
```

Comment:
subtracting 1 is not necessary. from the `snprintf` manual page:

> The snprintf() and vsnprintf() functions will write at most size-1 of the characters printed into the output string (the size'th character then gets the terminating `\0')

Reply:
removed -1 from the length field